### PR TITLE
Control thread pulling

### PR DESCRIPTION
**Pull request by sanderpick**
- Moves thread pulling control to config vars and flags.
- Makes logging level `INFO` by default instead of just `ERROR`.

### CODE REVIEW

File: common/common.go [hunk 1]
```diff
@@ -329,6 +347,17 @@ func WithNetLogstore(lt LogstoreType) NetOption {
 	}
 }
 
+func WithNetPulling(disabled bool, threadLimit int, startAfter, initialInterval, interval time.Duration) NetOption {
```
Comment (merlinran) [inline common/common.go:350]: It's a little surprising that `WithNetPulling(true, ...)` actually disables net pulling. I'd even vote for removing the `disabled` parameter entirely, so calling this simply enables pulling with the params.

File: net/net.go [hunk 2]
```diff
@@ -114,8 +104,29 @@ type net struct {
 
 // Config is used to specify thread instance options.
 type Config struct {
-	Debug  bool
-	PubSub bool
+	PullThreadsDisabled        bool
+	PullThreadsLimit           int
```
Comment (merlinran) [inline net/net.go:108]: Maybe make this `uint` to signal that it shouldn't be negative?

File: common/common.go (hunk 1 above)
Comment (merlinran) [reply common/common.go:350]: If pulling should be on by default, we could provide another utility called `NoNetPulling()` to disable it.

**Review by merlinran — APPROVED** [review]
LGTM with just some personal opinions

File: common/common.go (hunk 1 above)
Comment (sanderpick) [reply common/common.go:350]: Yeah, that is pretty weird now that you mention it :) I'll go with the `NoNetPulling` option.

**Pull request merged**